src/lib.rs

```diff
@@ -0,0 +1,419 @@
+extern crate core;
+
+pub mod command;
+
+use crate::command::{CargoCommand, Input};
+use cargo_metadata::{Message, MetadataCommand};
+use core::fmt;
+use rustc_version::Channel;
+use semver::Version;
+use serde::Deserialize;
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+use std::process::{Command, ExitStatus, Stdio};
+use std::{env, io, process};
+use tee::TeeReader;
+
+const DEFAULT_MESSAGE_FORMAT: &str = "json-render-diagnostics";
+
+/// Gets whether the executable should link the generated files to a 3ds
+/// based on the parsed input.
+pub fn get_should_link(input: &mut Input) -> bool {
+    // When running compile only commands, don't link the executable to the 3ds.
+    // Otherwise, link and run on the 3ds but do not run locally.
+    input.cmd == CargoCommand::Build
```

Comment:
This regressed. We shouldn't run 3dslink when doing a `build` command, only `run` and `test` (when not passing `--no-run`).

Reply:
Fixed to only return true for those two cases.